app/jobs/workflow_execution_submission_job.rb

```diff
@@ -33,9 +33,23 @@ def perform(workflow_execution)
     result = WorkflowExecutions::SubmissionService.new(workflow_execution, wes_connection).execute
 
     if result
-      WorkflowExecutionStatusJob.set(wait_until: 30.seconds.from_now).perform_later(workflow_execution.reload)
+      queue_job(workflow_execution)
     else
       handle_unable_to_process_job(workflow_execution, self.class.name)
     end
   end
+
+  def queue_job(workflow_execution)
+    min_run_time = case workflow_execution.workflow.settings.transform_keys(&:to_sym)
```

Comment:
Suggestion (blocking): Please move this min_run_time logic handling to be inside `lib/irida/pipeline.rb`

Reply:
Updated in 6c96662e5c897e3d15f316f6f4f34f90bd35d1b9